Bottom: c4f9e56fde30641b69e81060e3981b0df33f49c2
Top:    6c3a4d4cc0146d11acbd41ec30ac983f004e11c7
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2012-07-30 16:06:32 -0700

rspreload: Support dup2 calls

vsftpd requires dup2() support.  To handle dup2, we need to add
reference count tracking to the preload fd's.

Signed-off-by: Sean Hefty <sean.hefty@intel.com>


---

diff --git a/src/preload.c b/src/preload.c
index a680143..4b891a1 100644
--- a/src/preload.c
+++ b/src/preload.c
@@ -886,3 +886,10 @@ int fcntl(int socket, int cmd, ... /* arg */)
 	va_end(args);
 	return ret;
 }
+
+int dup2(int oldfd, int newfd)
+{
+	int fd;
+	return (fd_get(oldfd, &fd) == fd_rsocket) ?
+		: dup2(oldfd, newfd);
+}
